packages/blockchain-wallet-v4-frontend/src/data/signup/sagas.ts

```diff
@@ -314,12 +387,108 @@ export default ({ api, coreSagas, networks }) => {
       })
     )
   }
+  const pollForResetApproval = function* (sessionToken, n = 50) {
+    if (n === 0) {
+      yield put(actions.form.change(RECOVER_FORM, 'step', RecoverSteps.FORGOT_PASSWORD_EMAIL))
+      yield put(actions.alerts.displayInfo(C.VERIFY_DEVICE_EXPIRY, undefined, true))
+
+      return false
+    }
+    try {
+      yield delay(2000)
+      const response = yield call(api.pollForResetApprovalStatus, sessionToken)
+      // if (response?.status === 'true') {
+      if (response?.status === AccountRecoveryApprovalStatusType.APPROVED) {
+        yield put(actions.signup.setAccountRecoveryMagicLinkData(response))
+        yield put(actions.form.change(RECOVER_FORM, 'step', RecoverSteps.RECOVERY_OPTIONS))
+        return true
+      }
+      if (response?.status === AccountRecoveryApprovalStatusType.INVALID) {
+        yield put(actions.form.change(RECOVER_FORM, 'step', RecoverSteps.FORGOT_PASSWORD_EMAIL))
+        yield put(actions.alerts.displayError(C.VERIFY_DEVICE_FAILED, undefined, true))
+        return false
+      }
+    } catch (error) {
+      return false
+    }
+    return yield call(pollForResetApproval, sessionToken, n - 1)
```

Comment:
You might think about refactoring this to the redux-saga `retry` effect. https://redux-saga.js.org/docs/api/#retrymaxtries-delay-fn-args It might not be exactly what you're looking for though so just a suggestion.